core/trino-main/src/main/java/io/trino/sql/analyzer/Analysis.java

```diff
@@ -2058,7 +2058,7 @@ public TableFunctionInvocationAnalysis(
         {
             this.catalogHandle = requireNonNull(catalogHandle, "catalogHandle is null");
             this.functionName = requireNonNull(functionName, "functionName is null");
-            this.arguments = requireNonNull(arguments, "arguments is null");
+            this.arguments = ImmutableMap.copyOf(requireNonNull(arguments, "arguments is null"));
```

Comment:
requireNonNull not needed, since `ImmutableMap.copyOf` will not work with a null argument.